src/openssl/ssl/socket.cr

```diff
@@ -189,4 +189,15 @@ abstract class OpenSSL::SSL::Socket < IO
     io = @bio.io
     io.responds_to?(:remote_address) ? io.remote_address : nil
   end
+
+  {% for name in ["read_timeout", "read_timeout=", "write_timeout", "write_timeout="] %}
```

Comment:
Could you please define each method separately, without using macros? Not only because it's simpler, but only because this is wrong: `read_timeout=` expects an argument, same as `write_timeout=`. Thank you!